packages/react/accordion/src/Accordion.tsx

```diff
@@ -255,13 +261,30 @@ const AccordionImpl = React.forwardRef<AccordionImplElement, AccordionImplProps>
         case 'End':
           nextIndex = triggerCount - 1;
           break;
+        case 'ArrowRight':
+          if (orientation === 'horizontal') {
+            nextIndex = triggerIndex + 1;
+          }
+          break;
         case 'ArrowDown':
-          nextIndex = triggerIndex + 1;
+          if (orientation === 'vertical') {
+            nextIndex = triggerIndex + 1;
+          }
+          break;
+        case 'ArrowLeft':
+          if (orientation === 'horizontal') {
+            nextIndex = triggerIndex - 1;
+            if (nextIndex < 0) {
+              nextIndex = triggerCount - 1;
+            }
```

Comment:
Now that there's a couple places with this adjustement, it might make more sense to let the index run out of bounds negatively (as we do positively already) and cycle it afterwards where we are already clamping on line 292.

Reply:
Worth noting that the original `index` based implementation breaks when hiding triggers with css.

Not sure if it's worth doing now but should we eventually be using a [`focusFirst` approach](https://github.com/radix-ui/primitives/blob/81b25f4b40c54f72aeb106ca0e64e1e09655153e/packages/react/navigation-menu/src/NavigationMenu.tsx#L1125-L1137)?

Do we also need to support RTL with this change?

Reply:
Yeah, focusFirst probably in the end, although I think we can skip for now here (let us know later if you want to learn a bit more about what we're talking about here.)

RTL though yes that would make sense given we're now talking about horizontal orientation.

@dliu-workos, you'll need to use `useDirecttion` to get access to the normalized `dir` passed from context and adjust keyboard handling accordingly. We do this already in a bunch of components.